git-config/src/values.rs

```diff
@@ -577,18 +513,20 @@ impl Serialize for Integer {
     }
 }
 
-impl FromStr for Integer {
-    type Err = String;
```

Comment:
I think it's preferable to keep some error information, even if it's merely the original input that failed to parse as integer. In our case, it would have to be a `Vec<u8>` though, which isn't super convenient but better than nothing. Throughout `gitoxide` `bstr` is used, but it's not yet available in `git-config`. I certainly wouldn't mind to include it as dependency, but that could certainly be delayed to a later point in time.

Reply:
We will have quick error enums with things like IntergerParse error